### PR TITLE
Support 'not found' routes

### PR DESCRIPTION
**Pull request by mnmtanish**
- manually handle page.js callbacks array

### CODE REVIEW

File: client/_init.js [hunk 1]
```diff
@@ -3,6 +3,10 @@ FlowRouter = new Router;
 FlowRouter.Router = Router;
 FlowRouter.Route = Route;
 
+FlowRouter.notfound(function (params) {
```
Comment (arunoda) [inline client/_init.js]: Use a separate FlowRouter.notfound route for this.

File: client/_init.js [hunk 2]
```diff
@@ -3,6 +3,10 @@ FlowRouter = new Router;
...
+  console.error('# 404', params);
```
Comment (arunoda) [inline client/_init.js]: throw an error with the URL.

File: client/route.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 Route = function(router, path, options) {
+  var self = this;
```
Comment (arunoda) [inline client/route.js]: remove this.

File: client/router.js [hunk 4]
```diff
@@ -200,6 +205,45 @@ Router.prototype._invalidateTracker = function() {
...
+};
+
+
+Router.prototype._createCallback = function(path, handler) {
+  var route = new this._page.Route(path);
+  return route.middleware(handler);
+};
+
+
+Router.prototype._updateCallbacks = function() {
```
Comment (arunoda) [inline client/router.js]: use debounce.

**Comment by mnmtanish** [comment]
- Using a Route for 404
 - Debounced creating callbacks array

**Pull request merged**